### PR TITLE
CFY-4671 added reload of config roles

### PR DESCRIPTION
**Pull request by adamlavie**

### CODE REVIEW

File: flask_securest/authorization_providers/role_based_authorization_provider.py [hunk 1]
```diff
@@ -12,24 +12,56 @@
 #  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #  * See the License for the specific language governing permissions and
 #  * limitations under the License.
-
 import re
 import yaml
```
Comment (adamlavie) [inline flask_securest/authorization_providers/role_based_authorization_provider.py]: move to middle and above the watchdog

File: flask_securest/constants.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#########
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
```
Comment (adamlavie) [inline flask_securest/constants.py]: 2016

File: flask_securest/authorization_providers/role_based_authorization_provider.py [hunk 3]
```diff
@@ -12,24 +12,56 @@
...
+    def load_roles_config(self):
+        try:
+            with open(self.roles_config_file_path, 'r') as config_file:
+                self.permissions_by_roles = yaml.safe_load(config_file.read())
+                self.lgr.info('Loading of roles configuration ended '
+                              'successfully')
+        except (yaml.parser.ParserError, IOError) as e:
+            err = 'Failed parsing {role_config_file} file. Error: {error}.' \
+                .format(role_config_file=self.roles_config_file_path,
+                        error=str(e))
```
Comment (adamlavie) [inline flask_securest/authorization_providers/role_based_authorization_provider.py]: remove str

File: flask_securest/authorization_providers/role_based_authorization_provider.py [hunk 4]
```diff
@@ -12,24 +12,56 @@
...
+            self.lgr.warning(err)
+            raise ValueError(err)
+
+    def on_modified(self, event):
+        if os.path.abspath(event.src_path) == \
+                os.path.abspath(self.roles_config_file_path):
+            self.load_roles_config()
```
Comment (adamlavie) [inline flask_securest/authorization_providers/role_based_authorization_provider.py:62]: store var as abs from the get go

File: flask_securest/authorization_providers/role_based_authorization_provider.py [hunk 5]
```diff
@@ -12,24 +12,54 @@
...
         self.role_loader = role_loader
-        with open(roles_config_file_path, 'r') as config_file:
-            self.permissions_by_roles = yaml.safe_load(config_file.read())
+        self.permissions_by_roles = None
+        self.roles_config_file_path = os.path.abspath(roles_config_file_path)
+        self.observer = Observer()
+        self.observer.schedule(self,
+                               path=os.path.dirname(
+                                   os.path.abspath(roles_config_file_path)),
+                               recursive=False)
```
Comment (dankilman) [inline flask_securest/authorization_providers/role_based_authorization_provider.py:44]: while not reuse `self.roles_config_file_path`?

File: flask_securest/userstores/file_userstore.py [hunk 6]
```diff
@@ -40,7 +39,8 @@ def __init__(self, userstore_file_path):
         self.observer.start()
 
     def on_modified(self, event):
-        if event.src_path == self.userstore_file_path:
+        if os.path.abspath(event.src_path) == \
+                os.path.abspath(self.userstore_file_path):
```
Comment (dankilman) [inline flask_securest/userstores/file_userstore.py]: save as abspath

**Pull request merged**